app/file-upload/server/lib/FileUpload.js

```diff
@@ -265,6 +265,41 @@ export const FileUpload = {
 		return result;
 	},
 
+	createImageThumbnail(file) {
+		if (!settings.get('Message_Attachments_Thumbnails_Enabled')) {
+			return;
+		}
+
+		file = Uploads.findOneById(file._id);
+		file = FileUpload.addExtensionTo(file);
+		const store = FileUpload.getStore('Uploads');
+		const image = store._store.getReadStream(file._id, file);
+		const width = settings.get('Message_Attachments_Thumbnails_Width');
+		const height = settings.get('Message_Attachments_Thumbnails_Height');
+
+		const transformer = sharp()
```

Comment:
I was looking at sharp's documentation and found out about their `clone()` method (https://sharp.pixelplumbing.com/api-constructor#clone) and thought we might start using it, since we get a single upload and to many operation with it, we use it's metadata, create the "preview" and now the thumbnail. wdyt?

Reply:
But in that case, we'd need to create a "global" (at least, per upload scope) pipeline so we can `clone` it everywhere where needed 🤔.
Also, the operations done over the pipelines are quite different. For example, in `resizeImagePreview` and `createImageThumbnail` we share the resize, but with different options. Am I missing something? haha

Reply:
it would definitely require some additional/structural changes :grimacing: 

from what I understood it just shares a single input (in our case the uploaded file) so you can `clone()` and do different operations (or even do an operation that will be shared among clones - but I don't think we'd use this).. I *think* doing this way would be better (from a performance point of view) since we'd not need to create different streams and process them one by one.

Reply:
Ah I see. I think it would be a good addition. Instead of handling the buffer around, we can just use the same input stream and operate on it. That _should_ reduce memory consumption 🤔

Should we create a task for it 👀 to keep it on mind?

Reply:
> Should we create a task for it :eyes: to keep it on mind?

yep, pls do. thx

Reply:
Done!